basicwidget/popup.go

```diff
@@ -135,7 +143,7 @@ func (p *Popup) Open() {
 		return
 	}
 	if p.opacityCount > 0 {
-		p.Close()
+		p.close(PopupClosedReasonFuncCall)
```

Comment:
Hmm, I realized that this is neither `PopupClosedReasonFuncCall` nor `PopupClosedReasonClickOutside`. Could you add `PopupClosedReasonReopen` and use it?